### PR TITLE
cleanup changes to nuopc branch

### PR DESCRIPTION
**Pull request by mvertens**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

## PR checklist
- [x] Short (1 sentence) summary of your PR: 
    cleanup of #ifdefs and introduction of wrapper component that introduces stubs for nems 
- [x] Developer(s): 
    @mvertens, @DeniseWorthen
- [x] Suggest PR reviewers from list in the column to the right.
- [ ] Please copy the PR test results link or provide a summary of testing completed below.
    ERS_Vnuopc_Ld5.T62_g37.DTEST.cheyenne_intel passed successfully (hash upcoming for nuopc_dev)
- How much do the PR code changes differ from the unmodified code? 
    - [x] bit for bit (for just the simple test above)
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [x] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [x] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [x] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [x] Please provide any additional information or relevant details below:
Carried out only very limited testing - but @DeniseWorthen also carried out tests with NEMS, CMEPS and CICE6 that showed that these changes were bit for bit.
@dbailey - the following files came from changes introduced by @DeniseWorthen and were needed I believe to get cice6 working in NEMS with CMEPS - @DeniseWorthen could you please comment
```Fortran
cicecore/cicedynB/analysis/ice_history_shared.F90
cicecore/cicedynB/infrastructure/ice_read_write.F90
cicecore/cicedynB/infrastructure/ice_restart_driver.F90
cicecore/cicedynB/infrastructure/io/io_netcdf/ice_restart.F90

```

### CODE REVIEW

File: cicecore/cicedynB/infrastructure/ice_read_write.F90 [hunk 1]
```diff
@@ -1106,14 +1106,15 @@ subroutine ice_read_nc_xy(fid,  nrec,  varname, work,  diag, &
       integer (kind=int_kind) :: & 
          varid          , & ! variable id
          status             ! status output from netcdf routines
```
Comment (DeniseWorthen) [inline cicecore/cicedynB/infrastructure/ice_read_write.F90]: When uncommenting the 'optional diagnostics' in this routine, multiple declaration lines in this subroutine require adding an & at the end of the line. 

I believe a better solution would be to add the additional variables as separate lines which could be cleanly uncommented when diagnostic output is desired, eg:

!  integer (kind=int_kind) :: &
!  ndim, vnvar, & ! sizes of netcdf file
! id,                   & ! dimension index
! dimlin             & ! dimension size

File: cicecore/cicedynB/infrastructure/io/io_netcdf/ice_restart.F90 [hunk 2]
```diff
@@ -228,9 +227,7 @@ subroutine init_restart_write(filename_spec)
          call define_rest_field(ncid,'uvel',dims)
          call define_rest_field(ncid,'vvel',dims)
 
-#ifdef CESMCOUPLED
```
Comment (DeniseWorthen) [inline cicecore/cicedynB/infrastructure/io/io_netcdf/ice_restart.F90:231]: Adding coszen to the restart files replicates EMC_CICE5 behaviour.

File: cicecore/cicedynB/infrastructure/ice_restart_driver.F90 [hunk 3]
```diff
@@ -58,9 +58,7 @@ subroutine dumpfile(filename_spec)
           stressp_1, stressp_2, stressp_3, stressp_4, &
           stressm_1, stressm_2, stressm_3, stressm_4, &
           stress12_1, stress12_2, stress12_3, stress12_4
-#ifdef CESMCOUPLED
```
Comment (dabail10) [inline cicecore/cicedynB/infrastructure/ice_restart_driver.F90:61]: This still needs CESMCOUPLED around it as it is not used in the standalone CICE. Is there an equivalent ifdef for NEMS?

File: cicecore/cicedynB/infrastructure/io/io_netcdf/ice_restart.F90 [hunk 4]
```diff
@@ -84,7 +84,6 @@ subroutine init_restart_read(ice_ic)
          endif
          endif ! use namelist values if use_restart_time = F
 
-         write(nu_diag,*) 'Restart read at istep=',istep0,time,time_forc
```
Comment (dabail10) [inline cicecore/cicedynB/infrastructure/io/io_netcdf/ice_restart.F90:87]: Why are you deleting this line?

File: cicecore/cicedynB/infrastructure/io/io_netcdf/ice_restart.F90 (hunk 2 above)
Comment (dabail10) [reply cicecore/cicedynB/infrastructure/io/io_netcdf/ice_restart.F90:231]: Great news. We will still need some sort of ifdef around this.

File: cicecore/cicedynB/infrastructure/io/io_netcdf/ice_restart.F90 (hunk 4 above)
Comment (DeniseWorthen) [reply cicecore/cicedynB/infrastructure/io/io_netcdf/ice_restart.F90:87]: I saw that---I think this might have been an accidental delete.

**Review by dabail10 — CHANGES_REQUESTED** [review]
There are a couple things changed in the standalone code here. These will still need to be wrapped in some sort of ifdef.

File: cicecore/cicedynB/infrastructure/io/io_netcdf/ice_restart.F90 (hunk 2 above)
Comment (DeniseWorthen) [reply cicecore/cicedynB/infrastructure/io/io_netcdf/ice_restart.F90:231]: In the EMC CICE5, there was originally no coszen present as a restart field so I didn't realize this is a coupled-only field. There is no NEMSCOUPLED; do we need one or could we use the exisiting "coupled" ifdef?

**Comment by mvertens** [comment]
@dabail10 - @DeniseWorthen and I have addressed the issues you raised except for the cosine zenith angle being required on restart.

**Review by dabail10 — APPROVED** [review]
This will do for now. There is still a print statement being deleted in ice_restart.F90. Does EMC not use io_pio? Regardless, I am working on changes to add a "restart_coszen" flag.

**Pull request merged**